jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -166,4 +175,166 @@ public boolean setLoggerLevel(String loggerName, String levelName)
         jettyLogger.setLevel(level);
         return true;
     }
+
+    @Override
+    public Object getAttribute(String name) throws AttributeNotFoundException
+    {
+        Objects.requireNonNull(name, "Attribute Name");
+
+        switch (name)
+        {
+            case "LoggerNames":
+                return getLoggerNames();
+            case "LoggerCount":
+                return getLoggerCount();
+            default:
+                throw new AttributeNotFoundException("Cannot find " + name + " attribute in " + this.getClass().getName());
+        }
+    }
+
+    @Override
+    public void setAttribute(Attribute attribute) throws AttributeNotFoundException
+    {
+        Objects.requireNonNull(attribute, "attribute");
+        String name = attribute.getName();
+        // No attributes are writable
+        throw new AttributeNotFoundException("Cannot set attribute " + name + " because it is read-only");
+    }
+
+    @Override
+    public AttributeList getAttributes(String[] attributeNames)
+    {
+        Objects.requireNonNull(attributeNames, "attributeNames[]");
+
+        AttributeList ret = new AttributeList();
+        if (attributeNames.length == 0)
+            return ret;
+
+        for (String name : attributeNames)
+        {
+            try
+            {
+                Object value = getAttribute(name);
+                ret.add(new Attribute(name, value));
+            }
+            catch (Exception e)
+            {
+                // nothing much we can do, this method has no throwables, and we cannot use logging here.
+                e.printStackTrace();
+            }
+        }
+        return ret;
+    }
+
+    @Override
+    public AttributeList setAttributes(AttributeList attributes)
+    {
+        Objects.requireNonNull(attributes, "attributes");
```

Comment:
I'd tolerate `null` and do nothing.

Reply:
This is supposed to error out, as null is invalid.
All implementations within OpenJDK will throw an exception with null (different exceptions, but they all throw)

Reply:
`MBeanServerDelegateImpl` does not throw and returns an empty `AttributeList`.

Reply:
`MBeanServerDelegateImpl.getAttributes(String[] attributes)` returns all attributes when null is passed in.

Which is nice and all, but pointless.
As the lower level code in JMX would never even call this method if the `String[] attributes` is null.

``` java
MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
ObjectName objectName = ObjectName.getInstance("java.lang", "type", "OperatingSystem");

AttributeList attributeList = mbeanServer.getAttributes(objectName, null);
for (Attribute attr : attributeList.asList())
{
    System.out.printf("Attribute[%s] = %s%n", attr.getName(), attr.getValue());
}
```

results in ...

```
Exception in thread "main" javax.management.RuntimeOperationsException: Exception occurred trying to invoke the getter on the MBean
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttributes(DefaultMBeanServerInterceptor.java:660)
	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.getAttributes(JmxMBeanServer.java:705)
	at jmx.JmxAccessDemo.main(JmxAccessDemo.java:16)
Caused by: java.lang.IllegalArgumentException: Attributes cannot be null
	... 8 more
```

Reply:
`MBeanServerDelegateImpl.getAttributes(String[] attributes)` returns all attributes when null is passed in.

Which is nice and all, but pointless.
As the lower level code in JMX would never even call this method if the `String[] attributes` is null.

``` java
MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
ObjectName objectName = ObjectName.getInstance("java.lang", "type", "OperatingSystem");

AttributeList attributeList = mbeanServer.getAttributes(objectName, null);
for (Attribute attr : attributeList.asList())
{
    System.out.printf("Attribute[%s] = %s%n", attr.getName(), attr.getValue());
}
```

results in ...

```
Exception in thread "main" javax.management.RuntimeOperationsException: Exception occurred trying to invoke the getter on the MBean
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttributes(DefaultMBeanServerInterceptor.java:660)
	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.getAttributes(JmxMBeanServer.java:705)
	at jmx.JmxAccessDemo.main(JmxAccessDemo.java:16)
Caused by: java.lang.IllegalArgumentException: Attributes cannot be null
	... 8 more
```